data/mods/stadium/moves.ts

```diff
@@ -40,6 +40,32 @@ export const Moves: {[k: string]: ModdedMoveData} = {
 		inherit: true,
 		// FIXME: onBeforeMove() {},
 	},
+	haze: {
+		inherit: true,
+		onHit(target, source) {
+			this.add('-clearallboost');
+			for (const pokemon of this.getAllActive()) {
+				pokemon.clearBoosts();
+				if (pokemon.status !== 'tox') {
+					// This should cure the status of both Pokemon, and subsequently recalculate stats to remove the Paralysis/Burn Speed Drop.
+					pokemon.cureStatus();
+					target.recalculateStats!();
+				}
+				if (pokemon.status === 'tox') {
+					pokemon.setStatus('psn');
+				}
+				for (const id of Object.keys(pokemon.volatiles)) {
+					if (id === 'residualdmg') {
+						pokemon.volatiles[id].counter = 0;
```

Comment:
The video you linked seems to show the Tox status being fully cured by Haze, not turned into a weaker poison. Am I missing something?

Reply:
Ahhhh shit, you're right. I never noticed that when testing, I always assumed it was the same as RBY. That can be cut then.

Reply:
It’s weird how it seems to not be cured immediately, though.

Reply:
Yeah, I don't get that myself. Magnemite doesn't take the residual damage though, so the status is definitely removed in its entirety. Maybe the game can only update one of the graphics at a time or something? I dunno, I'm kinda spitballing here.